src/sentry/search/snuba/executors.py

```diff
@@ -324,6 +331,32 @@ def query(
                 # When its a simple django-only search, we count_hits like normal
                 return paginator.get_result(limit, cursor, count_hits=count_hits, max_hits=max_hits)
 
+        # This search is specific to Inbox. If we're using inbox sort and only querying
+        # postgres then we can use this sort method. Otherwise if we need to go to Snuba,
+        # fail.
+        if (
+            sort_by == "inbox"
+            and get_search_filter(search_filters, "for_review", "=")
+            # This handles tags and date parameters for search filters.
+            and not [
+                sf
+                for sf in search_filters
+                if sf.key.name not in self.postgres_only_fields.union(["date"])
+            ]
+        ):
+            group_queryset = group_queryset.extra(
+                select={"inbox_date": "sentry_groupinbox.date_added"},
+                tables=["sentry_groupinbox"],
+                where=[
+                    "sentry_groupinbox.group_id = sentry_groupedmessage.id",
```

Comment:
Does this use the new index? Should project be a filter here?

Reply:
Yeah we do need to filter this table by project for this to make sense, good catch!

Reply:
I'm not actually 100% sure this index will be used even with the project_id in place. I think it's worth trying out, but it might fail when we have multiple project? Will have to test in prod.

Reply:
Actually, I probably should filter this table by date as well...

Reply:
Ok, updated. Need to test out this query still, having some issues on the replica at the moment so will do later on

Reply:
Haven't been able to get this to be performant on orgs with a lot of inbox rows, but we'll merge for now to see if it's even useful on Sentry.